src/wagtail_vector_index/storage/models.py

```diff
@@ -220,6 +234,50 @@ def bulk_from_documents(
             seen_keys.add(key)
             yield objects_by_key[key]
 
+    async def abulk_from_documents(
```

Comment:
copy of the `bulk_from_documents` method above, but async

Reply:
This feels like a lot of business logic to duplicate, but I'm not sure how/if that's avoidable without making the code significantly less readable.